odoo/osv/expression.py

```diff
@@ -436,14 +436,14 @@ def select_from_where(cr, select_field, from_table, where_field, where_ids, wher
     res = []
     if where_ids:
         if where_operator in ['<', '>', '>=', '<=']:
-            cr.execute('SELECT "%s" FROM "%s" WHERE "%s" %s %%s' % \
+            cr.execute('SELECT DISTINCT "%s" FROM "%s" WHERE "%s" %s %%s' % \
                 (select_field, from_table, where_field, where_operator),
                 (where_ids[0],))  # TODO shouldn't this be min/max(where_ids) ?
             res = [r[0] for r in cr.fetchall()]
         else:  # TODO where_operator is supposed to be 'in'? It is called with child_of...
             for i in range(0, len(where_ids), cr.IN_MAX):
                 subids = where_ids[i:i + cr.IN_MAX]
-                cr.execute('SELECT "%s" FROM "%s" WHERE "%s" IN %%s' % \
+                cr.execute('SELECT DISTINCT "%s" FROM "%s" WHERE "%s" IN %%s' % \
                     (select_field, from_table, where_field), (tuple(subids),))
                 res.extend([r[0] for r in cr.fetchall()])
```

Comment:
We can still have duplicates across the chunks here, just not within a chunk.

Reply:
You are completely right!
Thanks for spotting that mistake.

Reply:
Shouldn't you still rely on DISTINCT to avoid the overhead of retrieving lots of duplicates over the network? In fact, is the use of a set() for second order duplicates really useful, considering we're already dropping by 3 order of magnitudes?

By the way, have we compared the performance of DISTINCT vs GROUPBY vs EXISTS here, in a real-life example? Gut feeling would say that DISTINCT is the most readable of the 3 and close to optimal performance, but it's worth double-checking.